In [26]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.preprocessing import LabelEncoder

In [27]:
data=pd.read_csv('traductions_2.csv',sep=';')

In [28]:
data.head(10)

client_id       name    prenom  telephone  traduction_id  \
0        546     REBAHI     AMINE  558500659            701   
1        547  BOUCHAFAA      AMAR  770130202            702   
2        543    DEGHOUL   MOHAMED  542764940            697   
3        544    MAHDADI     ZHOUR  550478415            698   
4        545       DINE     AHMED  554780334            700   
5        555   CHERIGUI    MOUNIR  550262263            709   
6        548     SEHILI      AMEL  555369007            703   
7        548     SEHILI      AMEL  555369007            703   
8        556       0809   slimane  774701881            710   
9        554      AHMED  BENYAHIA  776186142            712   

            created_at  id_document                 document  prix   langue  \
0  2021-10-02 14:14:38           25       Certificat Médical  1200  fr->arb   
1  2021-10-02 14:26:58           40     Registre de Commerce  5000  arb->fr   
2  2021-10-02 15:46:12          107       CONTRAT DE TRAVAIL  1400  fr->arb   
3  2021-10-03 13:14:34           28    Certificat De Travail  1000  arb->fr   
4  2021-10-03 14:09:26            1   attestation de travail  1500  arb->fr   
5  2021-10-04 07:30:00            4          Relevé de notes  1500  arb->fr   
6  2021-10-04 08:28:45           22  CERTIFICAT DE SCOLARITÉ  1500  fr->arb   
7  2021-10-04 08:28:45            4          Relevé de notes  1500  fr->arb   
8  2021-10-05 07:33:28            1   attestation de travail  1500  arb->fr   
9  2021-10-05 07:39:30            4          Relevé de notes  5000  arb->fr   

   nbcopie  
0        1  
1        1  
2        4  
3        1  
4        1  
5        2  
6        1  
7        1  
8        1  
9        1

In [29]:
data.shape

(675, 11)

We notice that we have 675 line and 11 columns

In [30]:
data.describe()

client_id     telephone  traduction_id  id_document          prix  \
count   675.000000  6.750000e+02     675.000000   675.000000    675.000000   
mean    784.354074  6.159748e+08     990.880000    28.771852   1609.111111   
std     137.798461  4.014811e+08     165.497081    29.975630   1122.825275   
min     543.000000  0.000000e+00     697.000000     1.000000    200.000000   
25%     670.500000  5.514109e+08     847.500000     7.000000   1200.000000   
50%     782.000000  5.598923e+08     989.000000    20.000000   1500.000000   
75%     905.500000  7.714361e+08    1134.500000    40.000000   1500.000000   
max    1025.000000  7.789849e+09    1274.000000   116.000000  20000.000000   

          nbcopie  
count  675.000000  
mean     3.068148  
std      5.420708  
min      1.000000  
25%      1.000000  
50%      1.000000  
75%      3.000000  
max     53.000000

In [31]:
y=data['langue']
y

0       fr->arb
1       arb->fr
2       fr->arb
3       arb->fr
4       arb->fr
         ...   
670     fr->arb
671     fr->arb
672     fr->arb
673    arb->ang
674     fr->arb
Name: langue, Length: 675, dtype: object

In [33]:
encoder=LabelEncoder()
encoder.fit_transform(y)

array([5, 3, 5, 3, 3, 3, 5, 5, 3, 3, 3, 3, 3, 1, 3, 2, 3, 3, 5, 5, 5, 5,
       5, 5, 5, 3, 3, 2, 5, 3, 5, 5, 3, 5, 3, 5, 5, 3, 5, 5, 5, 3, 2, 5,
       3, 3, 2, 3, 2, 2, 2, 3, 5, 3, 3, 5, 5, 5, 3, 5, 3, 5, 2, 5, 2, 5,
       5, 3, 5, 5, 5, 5, 2, 5, 0, 5, 3, 5, 2, 3, 5, 5, 3, 3, 5, 5, 5, 4,
       3, 5, 5, 5, 5, 5, 5, 5, 3, 5, 3, 3, 5, 2, 3, 5, 3, 4, 5, 5, 3, 3,
       5, 5, 5, 3, 5, 5, 5, 3, 3, 3, 3, 5, 3, 2, 5, 5, 5, 3, 3, 3, 5, 5,
       3, 5, 5, 3, 3, 3, 2, 3, 5, 2, 3, 3, 3, 2, 0, 1, 5, 5, 5, 5, 5, 5,
       3, 5, 5, 3, 5, 3, 5, 5, 5, 5, 5, 5, 3, 5, 5, 5, 5, 5, 3, 3, 5, 5,
       5, 3, 5, 5, 5, 5, 5, 5, 5, 3, 3, 5, 5, 2, 5, 5, 5, 5, 3, 3, 3, 5,
       5, 3, 3, 4, 3, 4, 3, 3, 5, 3, 3, 3, 5, 2, 2, 3, 5, 4, 5, 3, 5, 3,
       3, 3, 5, 5, 3, 3, 5, 3, 2, 3, 3, 3, 5, 4, 3, 3, 3, 3, 3, 5, 3, 5,
       3, 5, 5, 3, 3, 4, 3, 3, 3, 3, 3, 3, 5, 5, 5, 5, 3, 3, 3, 3, 5, 5,
       3, 3, 2, 3, 3, 3, 2, 5, 5, 5, 5, 5, 3, 3, 5, 3, 3, 3, 3, 3, 5, 0,
       2, 3, 5, 3, 3, 3, 3, 4, 4, 3, 3, 3, 5, 2, 4,